src/controller/task.ts

```diff
@@ -46,7 +47,7 @@ export default {
     try {
       const task: Task | null = await TaskModel.findById(req.params.id).lean()
       if (!task) {
-        return res.status(404).send({ message: 'task not found' })
+        return next(new HttpError(404, 'Task not found'))
```

Comment:
Better to have errors which describes type of the error.  Something like `TaskNotFoundError`. 

Having error like this means status code and error message is already handled inside.

So basically you will need to `throw new TaskNotFoundError()`. Since all code is wrapped in try/catch block, this error will be handled and passed to next middleware.

Reply:
possibly, fixed e1098e4418555c67b50397f454c93ff0a4312a66

Reply:
You are still using next() ... but possible to do  `throw new TaskNotFoundError()`

Reply:
fixed c612b1feb30118ac3c59790b282badd72380a0de